### ISSUE TITLE
DSTU2 Support

### ISSUE THREAD

**Issue by rbrush** [cerner/bunsen#1]
A lot of systems continue to use DSTU2, so we should add it as an option to Bunsen.

**Comment by boristyukin** [comment]
ideally to pass FHIR version number that will pick a proper HAPI model version

**Comment by rbrush** [comment]
Yeah, I think we can make it pretty easy to do that. There's a FhirVersionEnum from the HAPI API that would be a type-safe way for users to specify the version.

**Comment by boristyukin** [comment]
@rbrush have you thought about this more, Ryan? I am curious about use cases then someone needs to work with multiple vendors who are on different versions of FHIR. Another use case if one keeps historical bundles on one version and then need to upgrade to another version, without migrating archived bundles (which would be a real pain). These are common scenarios with analytics / DW projects.

One way is to build different versions of bunsen for a proper FHIR version. But I think ideally to pass an optional parameter to bunsen parser methods with FHIR version

**Comment by rbrush** [comment]
I think we can support this, with DSTU2, 3, and (soon) R4 datasets in the same process....similarly to how HAPI separates them. It will take some refactoring of the code but I'll dig a bit deeper over the next few days.

**Comment by boristyukin** [comment]
I like your solution and I am pretty sure most companies will move from
DSTU2 soon once R4 is out. So the concern is not DSTU2 itself, but how to
deal with FHIR version changes in the future, keeping in mind that one can
work with multiple versions at the same organization. Another use case is a
conversion of historical data in one format to another.

**Comment by rbrush** [comment]
Sounds good. The good news is that while working on this I managed to get most of the way to supporting R4, and am able to use both STU3 and R4 in a single Bunsen build in my local branch. I logged issue #21 to track the completion of that effort.

**Comment by boristyukin** [comment]
this is awesome, Ryan! you rock!!

**Comment by rbrush** [comment]
Related to this: we've released Bunsen 0.3.0, which supports both STU3 and early R4 builds. As part of this we moved some of the APIs into bunsen.stu3 packages to future proof usage, so users will need to switch. Some details here:

http://engineering.cerner.com/bunsen/fhir_versions.html

At this point I don't expect that we'll be able to easily add direct use for DSTU2, but the link above has some suggestions for converting that content to STU3 or R4.
